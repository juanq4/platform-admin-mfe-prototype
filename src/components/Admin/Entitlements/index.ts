export * from "./Entitlements.component";
export * from "./Entitlements.definition";
