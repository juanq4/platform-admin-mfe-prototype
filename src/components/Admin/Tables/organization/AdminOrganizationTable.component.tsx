import React, { memo, useMemo, useRef } from "react";
import { EntityTable, mapIdToCopyCell, mapIdToStatusCell, mapIdToTickerCell } from "../../EntityTable/";
import type { CopyCellProps, EntityTableColumnDef, StatusCellProps, TickerCellProps } from "../../EntityTable/";
import {
  AdminOrganizationTableDefault,
  AdminOrganizationsTableCellRenderer,
  AdminOrganizationsTableClassName,
  AdminOrganizationTableIdModel,
  AdminOrganizationsTableHeader,
  AdminOrganizationsTableFields,
} from "./AdminOrganizationTable.definition";
import type { AdminOrganizationTableProps } from "./AdminOrganizationTable.definition";

const AdminOrganizationTableBase = (props: AdminOrganizationTableProps): JSX.Element => {
  const { id, tableProps = {}, ...entityTableProps } = props;

  const columnDefs = useRef<EntityTableColumnDef[]>([
    {
      field: AdminOrganizationsTableFields.Name,
      headerName: AdminOrganizationsTableHeader.Name,
      minWidth: 220,
      flex: 1,
    },
    {
      field: AdminOrganizationsTableFields.Ticker,
      headerName: AdminOrganizationsTableHeader.Ticker,
      maxWidth: 128,
      minWidth: 128,
      cellRenderer: AdminOrganizationsTableCellRenderer.TickerCell,
    },
    {
      field: AdminOrganizationsTableFields.Active,
      headerName: AdminOrganizationsTableHeader.Status,
      maxWidth: 128,
      minWidth: 128,
      cellRenderer: AdminOrganizationsTableCellRenderer.StatusCell,
    },
    {
      field: AdminOrganizationsTableFields.Id,
      headerName: AdminOrganizationsTableHeader.Orgid,
      maxWidth: 330,
      minWidth: 330,
      cellRenderer: AdminOrganizationsTableCellRenderer.OrgIdCell,
    },
  ]);
  const idModel = useMemo(() => new AdminOrganizationTableIdModel(id), [id]);
  const { columnDefs: additionalColumns = [], components, ...restTableProps } = tableProps;

  const columns = useMemo(() => [...columnDefs.current, ...additionalColumns], [additionalColumns]);

  return (
    <div id={idModel.id} className={AdminOrganizationsTableClassName.Base}>
      <EntityTable
        id={idModel.entityTable.id}
        title={AdminOrganizationTableDefault.ToolbarTitle}
        tableProps={{
          columnDefs: columns,
          components: {
            [AdminOrganizationsTableCellRenderer.TickerCell]: (params: TickerCellProps) =>
              mapIdToTickerCell(params, idModel.tickerCellList),
            [AdminOrganizationsTableCellRenderer.OrgIdCell]: (params: CopyCellProps) =>
              mapIdToCopyCell(params, idModel.orgIdCellList),
            [AdminOrganizationsTableCellRenderer.StatusCell]: (params: StatusCellProps) =>
              mapIdToStatusCell(params, idModel.statusCellList),
            ...components,
          },
          ...restTableProps,
        }}
        {...entityTableProps}
      />
    </div>
  );
};

export const AdminOrganizationTable = memo(AdminOrganizationTableBase);
