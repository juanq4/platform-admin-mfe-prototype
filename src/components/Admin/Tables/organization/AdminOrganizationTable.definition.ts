import { IdModelBase, isNullOrWhiteSpace } from "@q4/nimbus-ui";
import type { Organization } from "../../../../definitions";
import type { OrganizationsQueryVariables } from "../../../../hooks";
import { CopyCellListId, EntityTableIdModel, StatusCellListId, TypeCellListId, TickerCellListId } from "../../EntityTable/";
import type { EntityTableProps } from "../../EntityTable/";

export enum AdminOrganizationsTableHeader {
  Name = "Organization Name",
  Ticker = "Ticker",
  Status = "Status",
  Orgid = "Org Id",
}

export enum AdminOrganizationsTableFields {
  Name = "name",
  Ticker = "identifiers",
  Active = "active",
  Id = "id",
}

export enum AdminOrganizationTableDefault {
  ToolbarTitle = "All organizations",
}

export enum AdminOrganizationsTableCellRenderer {
  TickerCell = "TickerCell",
  OrgIdCell = "OrgIdCell",
  StatusCell = "StatusCell",
  TypeCell = "TypeCell",
}

export enum AdminOrganizationsTableClassName {
  Base = "admin-organizations",
}

export type AdminOrganizationTableProps = Partial<EntityTableProps<Organization, OrganizationsQueryVariables["page"]>> &
  Pick<EntityTableProps<Organization, OrganizationsQueryVariables["page"]>, "items" | "error" | "loading" | "onError">;

export class AdminOrganizationTableIdModel extends IdModelBase {
  entityTable: EntityTableIdModel;
  orgIdCellList: CopyCellListId;
  statusCellList: StatusCellListId;
  typeCellList: TypeCellListId;
  tickerCellList: TickerCellListId;

  constructor(id: string) {
    super(id);

    if (isNullOrWhiteSpace(this.id)) {
      this.entityTable = new EntityTableIdModel(null);
      this.orgIdCellList = new CopyCellListId(null, null);
      this.statusCellList = new StatusCellListId(null, null);
      this.typeCellList = new TypeCellListId(null, null);
      this.tickerCellList = new TickerCellListId(null, null);

      return;
    }

    this.entityTable = new EntityTableIdModel(`${this.id}EntityTable`);
    this.orgIdCellList = new CopyCellListId(this.id, "OrgId-");
    this.statusCellList = new StatusCellListId(this.id);
    this.typeCellList = new TypeCellListId(this.id);
    this.tickerCellList = new TickerCellListId(this.id);
  }
}
