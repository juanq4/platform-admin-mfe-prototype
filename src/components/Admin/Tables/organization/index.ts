export * from "./AdminOrganizationTable.definition";
export * from "./AdminOrganizationTable.component";
