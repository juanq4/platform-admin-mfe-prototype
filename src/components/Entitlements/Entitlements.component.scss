@import "/src/styles/scss/variables";
@import "/src/styles/scss/mixins";

.organization-entitlements {

  .nui-tabs--v1 .nui-tabs_item--selected {
    pointer-events: all;
  }

  &_tabs {
    flex-grow: 1;

    .nui-tabs_item--selected {
      pointer-events: all;
    }
  }

  &_delete-button {
    margin-left: gux(1);
    opacity: 0;
    transition: opacity $transition-duration--default $transition-timing-function--default;
  }

  &_toolbar {
    width: 100%;

    &.nui-toolbar--v1 {
      .nui-toolbar {
        &_inner {
          min-height: inherit;
        }

        &_row {
          padding: 0;
        }
      }
    }
  }

  .nui-tabs_item:hover &_delete-button {
    opacity: 1;
  }
}
