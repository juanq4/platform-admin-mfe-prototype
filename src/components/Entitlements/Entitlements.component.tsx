import "./Entitlements.scss";
import type { SwapableProps } from "@q4/nimbus-ui";
import {
  Button,
  ButtonTheme,
  Message,
  PopoverMenu,
  PopoverMenuSize,
  PopoverMenuTheme,
  Swapable,
  Tabs,
  Toolbar,
  useVisibility,
} from "@q4/nimbus-ui";
import { Entitlement as EntitlementConstant } from "@q4/platform-definitions";
import { camelCase, startCase } from "lodash";
import type { ReactNode } from "react";
import React, { memo, useCallback, useMemo, useRef, useState } from "react";
import type { CamelCaseEntitlement } from "../../configurations/entitlement.configuration";
import { EntitlementLabel } from "../../configurations/entitlement.configuration";
import { FeatureFlag } from "../../configurations/feature.configuration";
// // import { useClaims } from "../../hooks/useClaims/useClaims.hook";
import { useFeatureFlags } from "../../hooks/useFeatureFlags/useFeatureFlags.hook";
import { isEngagementAnalyticsEntitlement, getDefaultEntitlementSelection } from "../../utils/entitlement/entitlement.utils";
import { getOrganizationDetailsMode } from "../../utils/organization/organization.utils";
import { OrganizationDetailsMode } from "../AdminContent/Organizations/Details/OrganizationDetails.definition";
import { AdminStudioSitesTable } from "../Tables/sites/AdminStudioSitesTable.component";
import { EngagementAnalyticsTierSelectorComponent } from "./EngagementAnalyticsTierSelector/EngagementAnalyticsTierSelector.component";
import type { EntitlementsProps, Entitlement } from "./Entitlements.definition";
import { EntitlementsClassName, EntitlementsIdModel, BaseEntitlementKeys } from "./Entitlements.definition";

const EntitlementsBase = (props: EntitlementsProps): JSX.Element => {
  const {
    id,
    organizationId,
    entitlements,
    orgSubdomain,
    sitesByOrgResp,
    onOrgSubdomainChange,
    onAdd,
    onRemove,
    getSitesByOrg,
  } = props;

  const [selectedEntitlement, setSelectedEntitlement] = useState(0);

  const entitlementToRemove = useRef<Entitlement>(null);

  const [menuVisible, handleMenuShow, handleMenuHide] = useVisibility();
  const [messageVisible, handleMessageShow, handleMessageHide] = useVisibility();
  const claims = useClaims();
  const flags = useFeatureFlags();

  const idModel = useMemo(() => new EntitlementsIdModel(id), [id]);

  const handleMenuClick = useCallback(
    (entitlement: Entitlement) => {
      onAdd(entitlement);
      handleMenuHide();
    },
    [onAdd, handleMenuHide],
  );

  function handleRemoveEntitlement(entitlement: Entitlement) {
    onRemove(entitlement);
    handleMessageHide();
  }

  const handleRemoveEntitlementClick = useCallback(
    (entitlement: Entitlement) => {
      entitlementToRemove.current = entitlement;
      handleMessageShow();
    },
    [handleMessageShow],
  );

  const handleEngagementAnalyticsTierSelection = useCallback(
    (tier: string) => {
      entitlements.forEach(
        (currentEntitlementTier) =>
          isEngagementAnalyticsEntitlement(currentEntitlementTier) && onRemove(currentEntitlementTier),
      );

      onAdd(tier);
    },
    [entitlements, onAdd, onRemove],
  );

  const tabs = useMemo(
    () =>
      entitlements?.sort().map((entitlement) => {
        const baseEntitlement = isEngagementAnalyticsEntitlement(entitlement)
          ? EntitlementConstant.EngagementAnalytics
          : entitlement;
        const camelCaseEntitlement = camelCase(baseEntitlement) as CamelCaseEntitlement;
        const label = startCase(baseEntitlement);

        return {
          id: idModel[`${camelCaseEntitlement}Tab`],
          label: (
            <>
              {label}
              {getOrganizationDetailsMode(claims.permissions, organizationId) === OrganizationDetailsMode.Edit && (
                <Button
                  className={EntitlementsClassName.DeleteButton}
                  id={idModel[`${camelCaseEntitlement}TabRemove`].id}
                  icon="q4i-trashbin-4pt"
                  theme={ButtonTheme.None}
                  onClick={() => handleRemoveEntitlementClick(entitlement)}
                />
              )}
            </>
          ),
          value: entitlement,
        };
      }),
    [entitlements, idModel, claims.permissions, organizationId, handleRemoveEntitlementClick],
  );

  const entitlementLayers = useMemo<SwapableProps["layers"]>(() => {
    const entitlementElements: { element: ReactNode; associatedEntitlement: Entitlement }[] = [
      {
        element: (
          <AdminStudioSitesTable
            id={idModel.sitesTable.id}
            onError={getSitesByOrg}
            sitesByOrgResp={sitesByOrgResp}
            orgSubdomain={orgSubdomain}
            onOrgSubdomainChange={onOrgSubdomainChange}
          />
        ),
        associatedEntitlement: EntitlementConstant.Studio,
      },
      {
        element: (
          <EngagementAnalyticsTierSelectorComponent
            id={idModel.engagementAnalyticsTierSelector.id}
            entitlements={entitlements}
            onTierSelect={handleEngagementAnalyticsTierSelection}
          />
        ),
        associatedEntitlement: EntitlementConstant.EngagementAnalytics,
      },
    ];

    return entitlements?.map((currentEntitlement) => {
      const baseEntitlement = isEngagementAnalyticsEntitlement(currentEntitlement)
        ? EntitlementConstant.EngagementAnalytics
        : currentEntitlement;

      return entitlementElements.find(
        (currentEntitlementElement) => currentEntitlementElement.associatedEntitlement === baseEntitlement,
      )?.element;
    });
  }, [
    entitlements,
    idModel.sitesTable.id,
    idModel.engagementAnalyticsTierSelector.id,
    getSitesByOrg,
    sitesByOrgResp,
    orgSubdomain,
    onOrgSubdomainChange,
    handleEngagementAnalyticsTierSelection,
  ]);

  function handleEntitlementTabChange(_index: number, value: string) {
    setSelectedEntitlement(tabs.findIndex((t) => t.value === value));
  }

  const entitlementOptions = useMemo(() => {
    let baseEntitlementKeys = BaseEntitlementKeys;

    if (!flags?.[FeatureFlag.MeetingScheduler]) {
      baseEntitlementKeys = baseEntitlementKeys.filter(
        (currentBaseEntitlement) => currentBaseEntitlement !== "MeetingScheduler",
      );
    }

    return baseEntitlementKeys.map((currentEntitlementKey) => {
      const defaultTierSelection = getDefaultEntitlementSelection(EntitlementConstant[currentEntitlementKey]);

      return {
        id: idModel[`menuAdd${currentEntitlementKey}`].id,
        label: EntitlementLabel[currentEntitlementKey],
        onClick: () => handleMenuClick(defaultTierSelection),
      };
    });
  }, [flags, idModel, handleMenuClick]);

  return (
    <div className={EntitlementsClassName.Base} id={idModel.id}>
      <Toolbar className={EntitlementsClassName.Toolbar}>
        <Tabs
          className={EntitlementsClassName.Tabs}
          items={tabs}
          selected={selectedEntitlement}
          onChange={handleEntitlementTabChange}
        />
        {getOrganizationDetailsMode(claims.permissions, organizationId) !== OrganizationDetailsMode.component && (
          <Button
            id={idModel.menuButton.id}
            label="Add Entitlement"
            icon="q4i-add-4pt"
            theme={ButtonTheme.Steel}
            onClick={handleMenuShow}
          />
        )}
        <PopoverMenu
          anchorTargetElementId={idModel.menuButton.id}
          id={idModel.menu.id}
          size={PopoverMenuSize.Small}
          theme={PopoverMenuTheme.White}
          visible={menuVisible}
          options={entitlementOptions}
          onCloseRequest={handleMenuHide}
        />
      </Toolbar>

      <Swapable selected={selectedEntitlement} layers={entitlementLayers} />

      <Message
        id={idModel.removeMessage.id}
        title="Remove Entitlement"
        message="Are you sure you want to remove this entitlement?"
        visible={messageVisible}
        targetElementId={idModel.id}
        primaryActionProps={{
          id: idModel.removeMessageConfirm.id,
          label: "Confirm",
          onClick: () => handleRemoveEntitlement(entitlementToRemove.current),
        }}
        secondaryActionProps={{ id: idModel.removeMessageCancel.id, label: "Cancel", onClick: handleMessageHide }}
        onCloseRequest={handleMessageHide}
      />
    </div>
  );
};

export const Entitlements = memo(EntitlementsBase);
