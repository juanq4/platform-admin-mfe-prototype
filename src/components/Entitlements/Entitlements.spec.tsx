import { Entitlement } from "@q4/platform-definitions";
import { mockFlags } from "jest-launchdarkly-mock";
import React from "react";
import { NimbusConfig } from "../../__mocks__/contexts/NimbusConfig.mock";
import { getAppWrapper } from "../../__mocks__/utils/wrappers";
import { FeatureFlag } from "../../configurations/feature.configuration";
import { getOrganizationDetailsMode } from "../../utils/organization/organization.utils";
import { fireEvent, render, screen } from "../../utils/testUtils";
import { OrganizationDetailsMode } from "../AdminContent/Organizations/Details/OrganizationDetails.definition";
import { Entitlements } from "./Entitlements.component";
import { EntitlementsIdModel } from "./Entitlements.definition";
import type { EntitlementsProps } from "./Entitlements.definition";

jest.mock("../../../contexts/session/useSession.hook");
const mockUseClaims = useClaims as jest.Mock;

jest.mock("../../../utils/organization/organization.utils");
const mockGetOrganizationDetailsMode = getOrganizationDetailsMode as jest.Mock;

describe("Entitlements Component", () => {
  const getSitesByOrg = jest.fn();
  const handleAdd = jest.fn();
  const handleRemove = jest.fn();
  const idModel = new EntitlementsIdModel("mockEntitlements");
  const studioEntitlement = Entitlement.Studio;

  const nullProps: EntitlementsProps = {
    organizationId: null,
    entitlements: null,
    sitesByOrgResp: null,
    orgSubdomain: null,
    onOrgSubdomainChange: null,
    getSitesByOrg: null,
    onAdd: null,
    onRemove: null,
  };

  const mockProps: EntitlementsProps = {
    id: idModel.id,
    organizationId: null,
    entitlements: [],
    sitesByOrgResp: null,
    orgSubdomain: null,
    onOrgSubdomainChange: null,
    getSitesByOrg,
    onAdd: handleAdd,
    onRemove: handleRemove,
  };

  const mockWithEntitlementProps = {
    ...mockProps,
    entitlements: [studioEntitlement],
  };

  function addEntitlement() {
    const menuButton = screen.getByTestId(idModel.menuButton.id);
    fireEvent.click(menuButton);
    const menu = screen.getByTestId(idModel.menu.id);
    expect(menu).toBeInTheDocument();
    const studioAddButton = screen.getByTestId(idModel.menuAddStudio.id);
    fireEvent.click(studioAddButton);
    expect(menu).not.toBeInTheDocument();
    expect(handleAdd).toBeCalledWith(studioEntitlement);
  }

  function removeButtonWithConfirmationModal() {
    const studioTab = screen.getByTestId(idModel.studioTab);
    const sitesTable = screen.getByTestId(idModel.sitesTable.id);
    expect(studioTab).toBeInTheDocument();
    expect(sitesTable).toBeInTheDocument();
    const trashButton = screen.getByTestId(idModel.studioTabRemove.id);
    fireEvent.click(trashButton);
    const messageModal = screen.getByTestId(idModel.removeMessage.id);
    expect(messageModal).toBeInTheDocument();
    return { messageModal, studioTab, sitesTable };
  }

  function removeEntitlement() {
    const { messageModal } = removeButtonWithConfirmationModal();
    const confirmButton = screen.getByTestId(idModel.removeMessageConfirm.id);
    fireEvent.click(confirmButton);
    expect(handleRemove).toBeCalledWith(studioEntitlement);
    expect(messageModal).not.toBeInTheDocument();
  }

  beforeAll(() => {
    mockGetOrganizationDetailsMode.mockReturnValue(OrganizationDetailsMode.Edit);
    mockUseClaims.mockReturnValue({});
  });

  test("7386002: [Given] no props are provided [Expect] the component to render", () => {
    render(
      <NimbusConfig.ConfigProvider>
        <Entitlements {...nullProps} />
      </NimbusConfig.ConfigProvider>,
      {
        container: getAppWrapper(),
      },
    );
    const rootElement = screen.queryByTestId("root");
    expect(rootElement).toMatchSnapshot();
  });

  test("1728217: [Given] the user tries to add the Studio entitlement [Expect] Studio tab and Sites table to appear", () => {
    const { rerender } = render(<Entitlements {...mockProps} />);
    addEntitlement();
    rerender(<Entitlements {...mockWithEntitlementProps} />);
    const studioTab = screen.getByTestId(idModel.studioTab);
    const sitesTable = screen.getByTestId(idModel.sitesTable.id);
    expect(studioTab).toBeInTheDocument();
    expect(sitesTable).toBeInTheDocument();
  });

  test("1728218: [Given] the user clicks the trash can on an existing Studio entitlement [Expect] message modal to appear [And] when Cancel button is clicked [Expect] modal to close and Studio tab and Sites table to still exist", () => {
    render(<Entitlements {...mockWithEntitlementProps} />);
    const { messageModal, studioTab, sitesTable } = removeButtonWithConfirmationModal();
    const cancelButton = screen.getByTestId(idModel.removeMessageCancel.id);
    fireEvent.click(cancelButton);
    expect(messageModal).not.toBeInTheDocument();
    expect(studioTab).toBeInTheDocument();
    expect(sitesTable).toBeInTheDocument();
  });

  test("1728219: [Given] the user clicks the trash can on an existing Studio entitlement [Expect] message modal to appear [And] when X (close button) is clicked [Expect] modal to close and Studio tab and Sites table to still exist", () => {
    render(<Entitlements {...mockWithEntitlementProps} />);
    const { messageModal, studioTab, sitesTable } = removeButtonWithConfirmationModal();
    const closeButton = screen.getByTestId(idModel.removeMessage.exit.id);
    fireEvent.click(closeButton);
    expect(messageModal).not.toBeInTheDocument();
    expect(studioTab).toBeInTheDocument();
    expect(sitesTable).toBeInTheDocument();
  });

  test("1728220: [Given] the user clicks the trash can on an existing Studio entitlement [Expect] message modal to appear [And] when Confirm button is clicked [Expect] modal to close and Studio tab and Sites table to be removed", () => {
    const { rerender } = render(<Entitlements {...mockWithEntitlementProps} />);
    const studioTab = screen.getByTestId(idModel.studioTab);
    const sitesTable = screen.getByTestId(idModel.sitesTable.id);
    removeEntitlement();
    rerender(<Entitlements {...mockProps} />);
    expect(studioTab).not.toBeInTheDocument();
    expect(sitesTable).not.toBeInTheDocument();
  });

  test("1737328: [Given] the user adds the Studio entitlement [And] then removes it [Expect] Studio tab and Sites table to appear and also be removed", () => {
    const { rerender } = render(<Entitlements {...mockProps} />);
    addEntitlement();
    rerender(<Entitlements {...mockWithEntitlementProps} />);
    const studioTab = screen.getByTestId(idModel.studioTab);
    const sitesTable = screen.getByTestId(idModel.sitesTable.id);
    expect(studioTab).toBeInTheDocument();
    expect(sitesTable).toBeInTheDocument();
    removeEntitlement();
    rerender(<Entitlements {...mockProps} />);
    expect(studioTab).not.toBeInTheDocument();
    expect(sitesTable).not.toBeInTheDocument();
  });

  test("5520487: [Given] an administrator is on the organization details page [And] The Meeting Scheduler feature flag is on [Expect] Meeting Scheduler to be an entitlement option", () => {
    mockFlags({
      [FeatureFlag.MeetingScheduler]: true,
    });

    render(<Entitlements {...mockProps} />);
    const menuButton = screen.getByTestId(idModel.menuButton.id);
    fireEvent.click(menuButton);
    const menu = screen.getByTestId(idModel.menu.id);
    expect(menu).toBeInTheDocument();

    const schedulerAddButton = screen.getByTestId(idModel.menuAddMeetingScheduler.id);
    expect(schedulerAddButton).toBeInTheDocument();
  });

  test("5520488: [Given] an administrator is on the organization details page [And] The Meeting Scheduler feature flag is off [Expect] Meeting Scheduler not to be an entitlement option", () => {
    mockFlags({
      [FeatureFlag.MeetingScheduler]: false,
    });

    render(<Entitlements {...mockProps} />);
    const menuButton = screen.getByTestId(idModel.menuButton.id);
    fireEvent.click(menuButton);
    const menu = screen.getByTestId(idModel.menu.id);
    expect(menu).toBeInTheDocument();

    const schedulerAddButton = screen.queryByTestId(idModel.menuAddMeetingScheduler.id);
    expect(schedulerAddButton).not.toBeInTheDocument();
  });
});
