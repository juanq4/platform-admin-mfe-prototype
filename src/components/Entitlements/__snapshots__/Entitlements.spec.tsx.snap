// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entitlements Component 7386002: [Given] no props are provided [Expect] the component to render 1`] = `
.emotion-0 {
  background-color: #ffffff;
  min-height: 64px;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
  transition-property: background-color,color;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 0 24px;
}

@media only screen and (max-width: 767px) {
  .emotion-1 {
    padding: 0 16px;
  }
}

.emotion-1 [data-type="toolbar-row"] {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 [data-type="toolbar-row"]:first-child {
  padding-top: 24px;
}

.emotion-1 [data-type="toolbar-row"]:last-child {
  padding-bottom: 24px;
}

.emotion-1 [data-type="toolbar-row"]:first-child:last-child {
  height: 100%;
}

.emotion-1 [data-type="toolbar-row"]+.emotion-1 [data-type="toolbar-row"] {
  padding-top: 8px;
}

.emotion-1 [data-type="toolbar-row"]>*:not(:first-child) {
  margin-left: 8px;
}

.emotion-1 [data-type="toolbar-row"][class*="--justify-flex-start"] {
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

.emotion-1 [data-type="toolbar-row"][class*="--justify-center"] {
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 [data-type="toolbar-row"][class*="--justify-center-space-between"] {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-1 [data-type="toolbar-row"][class*="--justify-center-space-around"] {
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
}

.emotion-1 [data-type="toolbar-row"][class*="--justify-center-flex-end"] {
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
}

@media only screen and (max-width: 412px) {
  .emotion-1 [data-type="toolbar-row"]:first-child {
    padding-top: 16px;
  }

  .emotion-1 [data-type="toolbar-row"]:last-child {
    padding-bottom: 16px;
  }

  .emotion-1 [data-type="toolbar-row"]+.emotion-1 [data-type="toolbar-row"] {
    padding-top: 16px;
  }

  .emotion-1 [data-type="toolbar-row"]>*:not(:first-child) {
    margin-left: 16px;
  }
}

.emotion-1 [data-type="toolbar-group"]>*:not(:first-child) {
  margin-left: 8px;
}

@media only screen and (max-width: 767px) {
  .emotion-1 [data-type="toolbar-group"]>*:not(:first-child) {
    margin-left: 16px;
  }
}

.emotion-1 [data-type="toolbar-group"][class*="--text"]>*:not(:first-child) {
  margin-left: 16px;
}

.emotion-1[class*="_title "] {
  padding: 4px 0;
}

.emotion-1[class*="_annotation "] {
  display: block;
  font-size: 10px;
  line-height: 12px;
  margin-top: 4px;
}

.emotion-2 {
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  padding: 2px 0 2px 2px;
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 0;
  height: 36px;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  border-radius: 18px;
  background-color: #cce4fa;
}

@media only screen and (max-width: 768px) {
  .emotion-2 {
    height: 40px;
    border-radius: 20px;
  }
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  background-color: #0f5ca3;
  color: #ffffff;
  border-color: transparent;
  font-family: inherit;
  height: 40px;
  padding: 12px 16px;
  border-width: 1px;
  border-style: solid;
  font-size: 14px;
  line-height: 16px;
  text-transform: none;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
  cursor: pointer!important;
  border-radius: 4px;
}

.emotion-3:hover {
  background: #0a3d6d;
  color: #ffffff;
}

.emotion-3:hover [data-type="nui-tooltip"] {
  opacity: 1;
  visibility: visible;
}

.emotion-3:focus {
  background: #0a3d6d;
  border-color: #4ea1ef;
  box-shadow: 0px 0px 3px #4ea1ef;
}

.emotion-3:active {
  background: #051f36;
}

.emotion-3:disabled {
  pointer-events: none;
  cursor: default;
  background: #f7fafc;
  color: #61707c;
  border-color: transparent;
}

@media only screen and (max-width: 768px) {
  .emotion-3 {
    height: 40px;
    padding: 12px;
  }
}

.emotion-4 {
  font-size: 12px;
}

.emotion-4+span {
  margin-left: 4px;
}

@media only screen and (max-width: 768px) {
  .emotion-4 {
    line-height: 16px;
  }
}

.emotion-5 {
  display: inline-block;
  cursor: pointer;
}

<div
  id="root"
>
  <div
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="-1"
  />
  <div
    data-focus-lock-disabled="disabled"
  />
  <div
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="-1"
  />
  <div
    class="organization-entitlements"
  >
    <section
      class="organization-entitlements_toolbar emotion-0"
    >
      <div
        class="organization-entitlements_toolbar_inner emotion-1"
      >
        <div
          class="nui-toolbar_row nui-toolbar_row--justify-space-between"
          data-type="toolbar-row"
        >
          <ul
            class="organization-entitlements_tabs emotion-2"
          />
          <button
            class="emotion-3"
            type="button"
          >
            <i
              class="q4i-add-4pt emotion-4"
            />
            <span
              class="emotion-5"
            >
              Add Entitlement
            </span>
          </button>
        </div>
      </div>
    </section>
    <div
      class="nui-swapable"
    />
  </div>
</div>
`;
